### PR TITLE
Update Google Signin library to v4

### PR DESCRIPTION
**Pull request by cocojoe**
Removed retired Google SignIn code
Updated tests
Cleaned up podfile

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,9 +1,10 @@
 language: objective-c
-osx_image: xcode7
+osx_image: xcode8.2
 before_install:
 - gem install cocoapods xcpretty --no-rdoc --no-ri --no-document
+- pod repo update
 script:
-- set -o pipefail && xcodebuild test -workspace LockGoogle.xcworkspace -scheme LockGoogle -sdk iphonesimulator ONLY_ACTIVE_ARCH=NO | xcpretty -c
+- set -o pipefail && xcodebuild test -workspace LockGoogle.xcworkspace -scheme LockGoogle -destination 'id=AD8E8C76-910E-437D-88C2-4D6F6EBE3355' | xcpretty -c
```
Comment (hzalaz) [inline .travis.yml:7]: I believe the ids can change from installation to installation so we should use names instead

File: LockGoogleApp/Info.plist [hunk 2]
```diff
@@ -3,9 +3,9 @@
 <plist version="1.0">
 <dict>
 	<key>Auth0ClientId</key>
-	<string>U5MhUrbyQHSVWjlEqZSTCBUFABLbJAS3</string>
+	<string>YN9SCHpqf8uLRpZbkLfehgmqlAObPgXL</string>
 	<key>Auth0Domain</key>
-	<string>overmind.auth0.com</string>
+	<string>brucke.auth0.com</string>
```
Comment (hzalaz) [inline LockGoogleApp/Info.plist]: We should not commit clientid of this tenant (removed it from dashboard already) so put back the old ones

**Review by hzalaz — CHANGES_REQUESTED** [review]

File: LockGoogleApp/Info.plist (hunk 2 above)
Comment (cocojoe) [reply LockGoogleApp/Info.plist]: No problem, I wasn't sure, should have double checked with you.  The intention was to leave it as a working example (previous one wasn't).

File: .travis.yml (hunk 1 above)
Comment (cocojoe) [reply .travis.yml:7]: I went through many iterations as the names setup just wouldn't work (see travis PR history). There is an ongoing travis issue and the 'solution' for now is to specify id.

https://discuss.circleci.com/t/multiple-ios-simulators-available-for-single-os-device-configurations/7854

**Review by hzalaz — APPROVED** [review]

**Pull request merged**